### PR TITLE
Remove bluebird dependency

### LINKED ISSUE #217: Core: rm bluebird dependency
it has very big uninformative stack traces that complicate debugging. 
 custom 
and it is not possible to switch to another lib once start using. a lot of wrappers around standard then\catch

### PR DESCRIPTION
**Pull request by stanislaw-glogowski**
connect #217

### CODE REVIEW

**Comment by stanislaw-glogowski** [comment]
Same tests fail on master branch

File: lib/services/tokens/token.dao.js [hunk 1]
```diff
@@ -72,27 +69,31 @@ dao.find = function (tokenObj, options) {
...
-            if (!token) {
-              return Promise.reject(new Error());
-            }
+      const getTokenPromise = new Promise((resolve, reject) => {
+        activeTokenIds.reverse();
 
-            isEqual = Object.keys(tokenObj).every((key) => tokenObj[key] === token[key]);
-            return isEqual ? token : Promise.reject(new Error());
-          });
+        const next = () => {
```
Comment (DrMegavolt) [inline lib/services/tokens/token.dao.js]: I believe you have correct assumption that typically there are not many active tokens, most likely 0-1. So this approach should work

File: lib/services/tokens/token.dao.js [hunk 2]
```diff
@@ -57,7 +54,7 @@ dao.find = function (tokenObj, options) {
 
   return db.hgetallAsync(consumerTokensKey)
     .then((tokenIds) => {
-      let tokenPromises, activeTokenIds, getTokenPromise;
+      let activeTokenIds;
```
Comment (XVincentX) [inline lib/services/tokens/token.dao.js]: We need a style guide, or we will keep changing these lines of code according to the tastes of the person that's working on it.

File: lib/services/tokens/token.dao.js [hunk 3]
```diff
@@ -72,27 +69,31 @@ dao.find = function (tokenObj, options) {
...
-          });
+        const next = () => {
+          if (!activeTokenIds.length) {
+            reject(new Error());
+            return;
+          }
+          db.hgetallAsync(tokenNamespace.concat(':', activeTokenIds.pop()))
+            .then((token) => {
+              if (!token || !Object.keys(tokenObj).every((key) => tokenObj[key] === token[key])) {
+                throw new Error();
```
Comment (XVincentX) [inline lib/services/tokens/token.dao.js]: I'd put an error message here.

File: lib/services/tokens/token.dao.js [hunk 4]
```diff
@@ -155,11 +156,11 @@ dao.revoke = function (token, options) {
...
-        .multi()
-        .hset(tokenNamespace.concat(':', token.id), 'archived', 'true')
-        .hdel(consumerTokensNamespace.concat(':', token.consumerId), token.id)
-        .hset(consumerTokensExpiredNamespace.concat(':', token.consumerId), token.id, 'true')
-        .execAsync();
+    .multi()
+    .hset(tokenNamespace.concat(':', token.id), 'archived', 'true')
+    .hdel(consumerTokensNamespace.concat(':', token.consumerId), token.id)
+    .hset(consumerTokensExpiredNamespace.concat(':', token.consumerId), token.id, 'true')
+    .execAsync();
```
Comment (XVincentX) [inline lib/services/tokens/token.dao.js]: dtto for the style guide comment

File: lib/services/utils.js [hunk 5]
```diff
@@ -1,51 +1,57 @@
...
-  return bcrypt.genSalt(config.systemConfig.crypto.saltRounds)
-  .then(function (salt) {
-    return bcrypt.hashAsync(password, salt);
-  })
-  .catch(() => Promise.reject(new Error('password hash failed'))); // TODO: replace with server error
+
+  return bcrypt
+    .genSalt(config.systemConfig.crypto.saltRounds)
+    .then((salt) => hashAsync(password, salt))
+    .catch(() => Promise.reject(new Error('password hash failed'))); // TODO: replace with server error
```
Comment (XVincentX) [inline lib/services/utils.js]: This feels really weird (not your thing - I just was having a look)

File: lib/services/tokens/token.dao.js (hunk 2 above)
Comment (stanislaw-glogowski) [reply lib/services/tokens/token.dao.js]: I didn't want to change logic / style of existing code. I only removed `bluebird` promise library.

File: lib/services/tokens/token.dao.js (hunk 3 above)
Comment (stanislaw-glogowski) [reply lib/services/tokens/token.dao.js]: new Error() was here before. Take a look what happened with that error next (`catch(() => null)`). In my opinion whole file needs to be rewritten.

**Comment by DrMegavolt** [comment]
@stanislaw-glogowski @XVincentX while there is a general code style\architecture problem in this module please make as less changes as possible. 

This is far more complicated than  Error() vs Error(message) or  const vs let. Once you get good understanding of what this service is doing in combination with other consumer management parts we can start discussing changes 

And the way to understand is JWT implementation #419 #30

**Pull request merged**